widgets/src/lib/InferenceWidget/widgets/ImageToTextWidget/ImageToTextWidget.svelte

```diff
@@ -0,0 +1,99 @@
+<script>
+	import type { WidgetProps } from "../../shared/types";
+
+	import WidgetDropzone from "../../shared/WidgetDropzone/WidgetDropzone.svelte";
+	import WidgetOutputText from "../../shared/WidgetOutputText/WidgetOutputText.svelte";
+	import WidgetWrapper from "../../shared/WidgetWrapper/WidgetWrapper.svelte";
+	import { getResponse } from "../../shared/helpers";
+
+	export let apiToken: WidgetProps["apiToken"];
+	export let apiUrl: WidgetProps["apiUrl"];
+	export let model: WidgetProps["model"];
+	export let noTitle: WidgetProps["noTitle"];
+
+	let computeTime = "";
+	let error: string = "";
+	let isLoading = false;
+	let modelLoading = {
+		isLoading: false,
+		estimatedTime: 0,
+	};
+	let output = "";
+	let outputJson: string;
+
+	function onSelectFile(file: File) {
+		getOutput(file);
+	}
+
+	async function getOutput(file: File, withModelLoading = false) {
+		if (!file) {
+			return;
+		}
+
+		const requestBody = { file };
+
+		isLoading = true;
+
+		const res = await getResponse(
+			apiUrl,
+			model.modelId,
+			requestBody,
+			apiToken,
+			parseOutput,
+			withModelLoading
+		);
+
+		isLoading = false;
+		// Reset values
+		computeTime = "";
+		error = "";
+		modelLoading = { isLoading: false, estimatedTime: 0 };
+		output = "";
+		outputJson = "";
+
+		if (res.status === "success") {
+			computeTime = res.computeTime;
+			output = res.output;
+			outputJson = res.outputJson;
+		} else if (res.status === "loading-model") {
+			modelLoading = {
+				isLoading: true,
+				estimatedTime: res.estimatedTime,
+			};
+			getOutput(file, true);
+		} else if (res.status === "error") {
+			error = res.error;
+		}
+	}
+
+	function parseOutput(body: unknown): string {
+		if (Array.isArray(body) && body.length) {
+			const firstEntry = body[0];
+			return (
+				firstEntry["generated_text"] ?? // text-generation + text2text-generation + image-to-text
+				firstEntry["translation_text"] ?? // translation
```

Comment:
Since we know here that we're in the case of `image-to-text`, why not just check for `firstEntry["generated_text"]`? I think that the behavior in `TextGenerationWidget` is because it can handle different tasks, which is not the case here? (cc @beurkinger, I might be wrong!)

Reply:
done 👍